### PR TITLE
SECS-3039: Migrated secrets and runners to CRT

### PR DESCRIPTION
**Pull request by amhashicorp**
## TLDR
Migrating secrets to CRT vault and runners to self hosted.

## Description
Migrating secrets to CRT vault and runners to self hosted to meet required company standards

## Fixes
https://hashicorp.atlassian.net/browse/SECS-3039

## Type of change
GHA config

## How Has This Been Tested?
GHA runs will complete on PR

## PCI review checklist

- [ ] If applicable, I’ve documented a plan to revert these changes if they require more than reverting the pull request.

- [ ] If applicable, I’ve worked with GRC to document the impact of any changes to security controls.

  Examples of changes to controls include access controls, encryption, logging, etc.

- [ ] If applicable, I’ve worked with GRC to ensure compliance due to a significant change to the cardholder data environment.

  Examples include changes to operating systems, ports, protocols, services, cryptography-related components, PII processing code, etc.

If you have any questions, please contact your direct supervisor, GRC (#team-grc), or the PCI working group (#proj-pci-core). You can also find more information at [PCI Compliance](https://hashicorp.atlassian.net/wiki/spaces/SEC/pages/2784559202/PCI+Compliance).

### CODE REVIEW

**Comment by amhashicorp** [comment]
Closing this PR as the changes are not applicable due to the repo being public.
